examples/petset/mongodb-petset-persistent.yaml

```diff
@@ -0,0 +1,141 @@
+    value: sampledb
+    required: true
+
+  - name: MONGODB_ADMIN_PASSWORD
+    displayName: "MongoDB Admin Password"
+    description: "Password for the database admin user."
+    generate: expression
+    from: "[a-zA-Z0-9]{16}"
+    required: true
+
+  - name: MONGODB_REPLICA_NAME
+    displayName: "Replica Set Name"
+    description: "The name of the replica set."
+    value: rs0
+    required: true
+
+  - name: MONGODB_KEYFILE_VALUE
+    displayName: "Keyfile Content"
+    description: "The value of the MongoDB keyfile (https://docs.mongodb.com/manual/core/security-internal-authentication/#internal-auth-keyfile)."
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
+    required: true
+
+  - name: MONGODB_IMAGE
+    displayName: "MongoDB Docker Image"
+    description: "A reference to a supported MongoDB Docker image (centos/mongodb-32-centos7, centos/mongodb-26-centos7, openshift/mongodb-24-centos7, ...)."
+    value: "centos/mongodb-32-centos7"
+    required: true
+
+  - name: MONGODB_SERVICE_NAME
+    displayName: "OpenShift Service Name"
+    description: "The name of the OpenShift Service exposed for the database."
+    value: mongodb
+    required: true
+
+  - name: VOLUME_CAPACITY
+    displayName: "Volume Capacity"
+    description: "Volume space available for data, e.g. 512Mi, 2Gi."
+    value: "1Gi"
+    required: true
+
+objects:
+  # A headless service to create DNS records
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: "${MONGODB_SERVICE_NAME}"
+    spec:
+      clusterIP: None
+      # the list of ports that are exposed by this service
+      ports:
+        - name: mongodb
+          port: 27017
+      # will route traffic to pods having labels matching this selector
+      selector:
+        name: "${MONGODB_SERVICE_NAME}"
+
+  - kind: PetSet
+    apiVersion: apps/v1alpha1
+    metadata:
+      name: mongodb
+    spec:
+      # pets get DNS/hostnames that follow the pattern: ${metadata.name}-NUM.${spec.serviceName}.default.svc.cluster.local
+      serviceName: "${MONGODB_SERVICE_NAME}"
+      replicas: 3
+      # describes the pod that will be created if insufficient replicas are detected
+      template:
+        metadata:
+          # this label will be used for count running pods
+          labels:
+            name: "${MONGODB_SERVICE_NAME}"
+          annotations:
+            # 'false' pauses a petset after creation of each pet, to avoid it we set it to 'true'
+            pod.alpha.kubernetes.io/initialized: "true"
+        spec:
+          containers:
+            - name: mongo-container
+              image: "${MONGODB_IMAGE}"
+              # list of ports to expose from the container
+              ports:
+                - containerPort: 27017
+              args:
+                - "run-mongod-pet"
+              volumeMounts:
+                - name: mongo-data
+                  mountPath: "/var/lib/mongodb/data"
+              redinessProbe:
```

Comment:
Just found with `git bisect` that after fixing readiness probe all things stopped working :-( It took too much time (20 sec -> 1 min) and at the end there is only one node in replica set and I see errors in the secondaries logs:
```
=> [Tue Oct 11 12:49:40] Adding mongodb-1.mongodb.myproject.svc.cluster.local to replica set ...
2016-10-11T12:49:40.233+0000 I NETWORK  [thread1] Starting new replica set monitor for rs0/mongodb-0.mongodb.myproject.svc.cluster.local:27017
2016-10-11T12:49:40.234+0000 I NETWORK  [ReplicaSetMonitorWatcher] starting
{
		"ok" : 0,
		"errmsg" : "Quorum check failed because not enough voting nodes responded; required 2 but only the following 1 voting nodes responded: mongodb-0.mongodb.myproject.svc.cluster.local:27017; the following nodes did not respond affirmatively: mongodb-1.mongodb.myproject.svc.cluster.local:27017 failed with HostUnreachable",
		"code" : 74
}
=> [Tue Oct 11 12:49:40] ERROR: couldn't join replica set!
```
and primary logs:
```
2016-10-12T10:16:15.842+0000 I REPL     [conn8] replSetReconfig admin command received from client
2016-10-12T10:16:15.845+0000 W NETWORK  [conn8] getaddrinfo("mongodb-1.mongodb.myproject.svc.cluster.local") failed: Name or service not known
2016-10-12T10:16:15.847+0000 I NETWORK  [conn8] getaddrinfo("mongodb-1.mongodb.myproject.svc.cluster.local") failed: Name or service not known
2016-10-12T10:16:15.847+0000 I REPL     [conn8] replSetReconfig config object with 2 members parses ok
2016-10-12T10:16:15.850+0000 W REPL     [ReplicationExecutor] Failed to complete heartbeat request to mongodb-1.mongodb.myproject.svc.cluster.local:27017; HostUnreachable: HostUnreachable
2016-10-12T10:16:15.850+0000 E REPL     [conn8] replSetReconfig failed; NodeNotFound: Quorum check failed because not enough voting nodes responded; required 2 but only the following 1 voting nodes responded: mongodb-0.mongodb.myproject.svc.cluster.local:27017; the following nodes did not respond affirmatively: mongodb-1.mongodb.myproject.svc.cluster.local:27017 failed with HostUnreachable
```
I'm putting it here for the record.